src/full/Agda/TypeChecking/Rules/LHS.hs

```diff
@@ -1667,7 +1668,7 @@ disambiguateConstructor ambC@(AmbQ cs) d pars = do
         -- but let's go with it for the sake of preserving the current behavior of Agda.
         -- Thus, only when 'constraintsOk' we require 'null disambs':
         -- (Note that in Haskell, boolean implication is '<=' rather than '=>'.)
-        (_, (c0,c,a) : disambs) | constraintsOk <= null disambs -> do
+        (_, [(c0,c,a)]) -> do
```

Comment:
Since you changed the code, you also need to update the comment.

Reply:
I removed the outdated comment and added/updated a few other comments.

Reply:
Thanks!